src/lib/components/dominant_color/index.svelte

```diff
@@ -61,6 +62,7 @@
 
 	$effect(() => {
 		if (image_loaded) return;
+		if (!is_intersecting) return;
```

Comment:
Yea LGTM, its working logic wise, I mean we cant confirm until we fix UI responsiveness to see effect on interface.
But this is ready to merge!